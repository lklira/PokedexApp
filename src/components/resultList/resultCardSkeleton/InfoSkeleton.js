import React, {useEffect, useRef} from 'react';
import {Animated} from 'react-native';

const InfoSkeleton = () => {
  const fadeAnim = useRef(new Animated.Value(0.3)).current;

  useEffect(() => {
    Animated.loop(
      Animated.sequence([
        Animated.timing(fadeAnim, {
          toValue: 0.8,
          duration: 700,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 0.3,
          duration: 700,
          useNativeDriver: true,
        }),
      ]),
    ).start();
  }, [fadeAnim]);

  return (
    <Animated.View
      style={{
        ...styles,
        opacity: fadeAnim,
      }}
    />
  );
};

const styles = {
  paddingVertical: 2,
  paddingHorizontal: 12,
  borderRadius: 16,
  marginBottom: 4,
  backgroundColor: '#fff',
  height: 16,
};

export default InfoSkeleton;
