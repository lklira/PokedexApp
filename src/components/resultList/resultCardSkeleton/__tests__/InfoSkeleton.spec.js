import React from 'react';
import {render} from '@testing-library/react-native';
import InfoSkeleton from '../InfoSkeleton.js';

describe('#InfoSkeleton', () => {
  let tree;
  beforeEach(() => {
    tree = render(<InfoSkeleton />);
  });

  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
