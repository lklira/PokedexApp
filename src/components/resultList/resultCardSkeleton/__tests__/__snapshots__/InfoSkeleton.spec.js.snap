// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#InfoSkeleton renders correctly 1`] = `
<View
  opacity={0.3}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 16,
      "height": 16,
      "marginBottom": 4,
      "opacity": 0.3,
      "paddingHorizontal": 12,
      "paddingVertical": 2,
    }
  }
/>
`;
